src/Neo/Network/P2P/Payloads/NotaryAssisted.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// NotaryAssisted.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.IO;
+using Neo.Json;
+using Neo.Persistence;
+using Neo.SmartContract.Native;
+using System.IO;
+using System.Linq;
+
+namespace Neo.Network.P2P.Payloads
+{
+    public class NotaryAssisted : TransactionAttribute
+    {
+        /// <summary>
+        /// Native Notary contract hash stub used until native Notary contract is properly implemented.
+        /// </summary>
+        private static readonly UInt160 notaryHash = Neo.SmartContract.Helper.GetContractHash(UInt160.Zero, 0, "Notary");
+
+        /// <summary>
+        /// Indicates the number of keys participating in the transaction (main or fallback) signing process.
+        /// </summary>
+        public byte NKeys;
+
+        public override TransactionAttributeType Type => TransactionAttributeType.NotaryAssisted;
+
+        public override bool AllowMultiple => false;
+
+        public override int Size => base.Size + sizeof(byte);
+
+        protected override void DeserializeWithoutType(ref MemoryReader reader)
+        {
+            NKeys = reader.ReadByte();
+        }
+
+        protected override void SerializeWithoutType(BinaryWriter writer)
+        {
+            writer.Write(NKeys);
+        }
+
+        public override JObject ToJson()
+        {
+            JObject json = base.ToJson();
+            json["nkeys"] = NKeys;
+            return json;
+        }
+
+        public override bool Verify(DataCache snapshot, Transaction tx)
```

Comment:
So, only the Native Contract is able to use this `NotaryAssisted` attribute, right?

This is not related to the conversation on discord regarding the "delegated powered notary node"  signing?

Reply:
> So, only the Native Contract is able to use this NotaryAssisted attribute, right?

Only transaction that has native Notary contract as a signer.